lib/gui/app/styled-components.tsx

```diff
@@ -242,6 +247,9 @@ export const Alert = styled((props) => (
 
 export interface GenericTableProps<T> extends BaseTableProps<T> {
 	refFn: (t: BaseTable<T>) => void;
+	data: T[];
+	notAllSelected?: boolean;
```

Comment:
`notAllSelected` is calculated from other props so it should not be in props.

Reply:
It's used in the styled component's css which can only access the props. I can make another interface which extends it and adds only the `notAllSelected` property, but it seems a waste to me

Reply:
Move the logic calculating `notAllSelected` into the function that takes the props as parameters (`StyledTable`).

Reply:
```typescript
				${(props) => {
					const notAllSelected =
						props.multipleSelection &&
						props.checkedRowsNumber !== 0 &&
						props.checkedRowsNumber !== props.data.length;
					return notAllSelected
						? `
					font-weight: 600;
					color: ${colors.primary.foreground};
					background: ${colors.primary.background};

					::after {
						content: '–';
					}
					`
						: '';
				}}
```

Reply:
Right, good call, no need to have the `const ...` either, it's just a short if with the 3 props